cpp/scripts/include_checker.py

```diff
@@ -58,62 +107,100 @@ def list_all_source_file(file_regex, srcdirs):
                     all_files.append(src)
     return all_files
 
-def rel_include_warnings(dir, src, line_num, inc_file, top_inc_dirs):
-    warn = []
+
+def rel_include_warnings(dir, src, line_num, inc_file,
+                         top_inc_dirs) -> typing.List[Issue]:
+    warn: typing.List[Issue] = []
     inc_folders = inc_file.split(os.path.sep)[:-1]
     inc_folders_alt = inc_file.split(os.path.altsep)[:-1]
+
     if len(inc_folders) != 0 and len(inc_folders_alt) != 0:
-        w = "File: %s, line %d, warning: using %s and %s as path separators"
-        warn.append(w % (src, line_num, os.path.sep, os.path.altsep))
+        w = "using %s and %s as path separators" % (os.path.sep,
+                                                    os.path.altsep)
+        warn.append(Issue(False, w, src, line_num))
+
     if len(inc_folders) == 0:
         inc_folders = inc_folders_alt
+
     abs_inc_folders = [
         os.path.abspath(os.path.join(dir, *inc_folders[:i + 1]))
-        for i in range(len(inc_folders))]
+        for i in range(len(inc_folders))
+    ]
+
     if os.path.curdir in inc_folders:
-        w = ("File: %s, line %d, warning: rel include containing "
-             "reference to current folder '%s'")
-        warn.append(w % (src, line_num, os.path.curdir))
-    if any(reduce(operator.or_,
-                  [os.path.basename(p) == f for f in top_inc_dirs])
-           for p in abs_inc_folders):
-        w = "File: %s, line %d, warning: rel include going over %s folders"
-        warn.append(w % (src, line_num,
-                         "/".join("'" + f + "'" for f in top_inc_dirs)))
-    if (len(inc_folders) >= 3 and os.path.pardir in inc_folders and
-            any(p != os.path.pardir for p in inc_folders)):
-        w = ("File: %s, line %d, warning: rel include with more than "
+        w = "rel include containing reference to current folder '{}'".format(
+            os.path.curdir)
+        warn.append(Issue(False, w, src, line_num))
+
+    if any(
+            reduce(operator.or_,
```

Comment:
Is the reduce(operator.or_) just equivalent to `any`? Maybe any would be shorter

Reply:
Fixed